Code/EnginePlugins/RecastPlugin/Components/DetourCrowdAgentComponent.h

```diff
@@ -0,0 +1,138 @@
+#pragma once
+
+#include <Core/World/Component.h>
+#include <Core/World/World.h>
+#include <RecastPlugin/Components/AgentSteeringComponent.h>
+#include <RecastPlugin/RecastPluginDLL.h>
+
+class ezDetourCrowdWorldModule;
+class ezDetourCrowdAgentComponent;
+struct ezDetourCrowdAgentParams;
+
+//////////////////////////////////////////////////////////////////////////
+
+struct ezDetourCrowdAgentRotationMode
+{
+  using StorageType = ezUInt8;
+
+  enum Enum
+  {
+    LookAtNextPathCorner,
+    MatchVelocityDirection,
+    
+    Default = LookAtNextPathCorner
+  };
+};
+
+EZ_DECLARE_REFLECTABLE_TYPE(EZ_NO_LINKAGE, ezDetourCrowdAgentRotationMode);
+
+//////////////////////////////////////////////////////////////////////////
+
+class EZ_RECASTPLUGIN_DLL ezDetourCrowdAgentComponentManager : public ezComponentManager<ezDetourCrowdAgentComponent, ezBlockStorageType::FreeList>
+{
+  using SUPER = ezComponentManager<ezDetourCrowdAgentComponent, ezBlockStorageType::FreeList>;
+
+public:
+  explicit ezDetourCrowdAgentComponentManager(ezWorld* pWorld);
+  ~ezDetourCrowdAgentComponentManager();
+
+  virtual void Initialize() override;
+
+  ezDetourCrowdWorldModule* GetDetourCrowdWorldModule() const { return m_pDetourCrowdModule; }
+
+private:
+  void Update(const ezWorldModule::UpdateContext& ctx);
+
+  ezUInt32 m_uiNextOwnerId = 1;
+  ezDetourCrowdWorldModule* m_pDetourCrowdModule = nullptr;
+};
+
+//////////////////////////////////////////////////////////////////////////
+
+class EZ_RECASTPLUGIN_DLL ezDetourCrowdAgentComponent : public ezAgentSteeringComponent
+{
+  EZ_DECLARE_COMPONENT_TYPE(ezDetourCrowdAgentComponent, ezAgentSteeringComponent, ezDetourCrowdAgentComponentManager);
+
+  //////////////////////////////////////////////////////////////////////////
+  // ezComponent
+
+protected:
+  virtual void SerializeComponent(ezWorldWriter& stream) const override;
+  virtual void DeserializeComponent(ezWorldReader& stream) override;
+
+  //////////////////////////////////////////////////////////////////////////
+  // ezAgentSteeringComponent
+
+public:
+  ezDetourCrowdAgentComponent();
+  ~ezDetourCrowdAgentComponent();
+
+  virtual void SetTargetPosition(const ezVec3& vPosition) override;
```

Comment:
Please have a look which public functions and properties could use some documentation. Most of these functions may sound obvious, but often there are details that are useful to know, when one actually tries to use the component. If there are such details, it would be good to document them here (I usually do it on the setters). See other components for inspiration.